runtime/sam/expr/eval.go

```diff
@@ -669,7 +669,13 @@ func (i *Index) Eval(ectx Context, this super.Value) super.Value {
 
 func indexArrayOrSet(sctx *super.Context, ectx Context, inner super.Type, vector zcode.Bytes, index super.Value) super.Value {
 	id := index.Type().ID()
+	if super.IsUnsigned(id) {
+		index = LookupPrimitiveCaster(sctx, super.TypeInt64).Eval(ectx, index)
```

Comment:
I think you need to set id again: `id = index.Type().ID()`.

Reply:
right